mixer/pkg/config/crd/store.go

```diff
@@ -199,21 +143,40 @@ func (s *Store) Init(kinds []string) error {
 		return err
 	}
 	s.caches = make(map[string]cache.Store, len(kinds))
-	timeout := time.After(s.retryTimeout)
-	timeoutdone := make(chan struct{})
-	s.retryInterval = time.Second / 2
-	go func() {
-		<-timeout
-		close(timeoutdone)
-	}()
-	remainingKinds := s.checkAndCreateCaches(timeoutdone, d, lwBuilder, kinds)
-	if len(remainingKinds) > 0 {
-		// Wait asynchronously for other kinds.
-		go s.checkAndCreateCaches(s.donec, d, lwBuilder, remainingKinds)
+	s.informers = make(map[string]cache.SharedInformer, len(kinds))
+	remaining := s.checkAndCreateCaches(d, lwBuilder, kinds)
+	if len(remaining) > 0 {
+		log.Warnf("Failed to discover kinds: %v", remaining)
 	}
+
 	return nil
 }
 
+// WaitForSynced implements store.WaitForSynced interface
+func (s *Store) WaitForSynced(timeout time.Duration) error {
+	stop := time.After(timeout)
+	tick := time.NewTicker(10 * time.Millisecond)
+	defer tick.Stop()
+
+	for {
```

Comment:
can we just use `cache.WaitForCacheSync(stopCh, s.informers...)` ?

Reply:
I'd like to have timeout to propagate out. Not sure if the function does that. I actually just started from that function, but git does not detect the diff.

Reply:
👍 Just want to make sure we weren't accidentally duplicating stuff we had to pull in anyway.